### ISSUE TITLE
warning: empty trait list in `derive` should not emit if created by a macro

### ISSUE THREAD

**Issue by Havvy** [rust-lang/rust#54652]
The following code should not emit a warning at the call site of the macro:

```rust
#![crate_type="lib"]

macro_rules! ds {
    () => (#[derive()] struct StructDs;)
}

ds!{}
```

Currently, it does:

```
warning: empty trait list in `derive`
  --> src/lib.rs:8:12
   |
8  |     () => (#[derive()] struct StructDs;)
   |            ^^^^^^^^^^^
...
11 | ds!{}
   | ----- in this macro invocation
```

**Comment by fmease** [comment]
Since #79078, `#[derive()]` no longer triggers a lint *anywhere* / *at all*. Closing as (coincidentally) fixed.
